Zend/zend_opcode.c

```diff
@@ -153,6 +153,12 @@ ZEND_API void zend_function_dtor(zval *zv)
 		ZEND_ASSERT(function->common.function_name);
 		zend_string_release_ex(function->common.function_name, 1);
 
+		if (function->common.attributes) {
+			zend_hash_destroy(function->common.attributes);
+			free(function->common.attributes);
+			function->common.attributes = NULL;
+		}
```

Comment:
Isn't that needed for user defined functions as well (i.e. for the then-clause above)?

Reply:
```
<?php

function my_func(
	#[\SensitiveArgument] $foo
) {

}

$fn = new ReflectionFunction('my_func');
foreach ($fn->getParameters() as $param) {
	foreach ($param->getAttributes() as $attr) {
		var_dump($attr->getName());
	}
}
```

This doesn't leak for me, so probably "no".

```
==40265== Memcheck, a memory error detector
==40265== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==40265== Using Valgrind-3.15.0 and LibVEX; rerun with -h for copyright info
==40265== Command: sapi/cli/php test3.php
==40265== 
string(17) "SensitiveArgument"
==40265== 
==40265== HEAP SUMMARY:
==40265==     in use at exit: 0 bytes in 0 blocks
==40265==   total heap usage: 11,346 allocs, 11,346 frees, 1,682,654 bytes allocated
==40265== 
==40265== All heap blocks were freed -- no leaks are possible
==40265== 
==40265== For lists of detected and suppressed errors, rerun with: -s
==40265== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
```

Reply:
What about anonymous function included multiple times?

see my PR https://github.com/php/php-src/pull/7563

Reply:
```
		/* op_arrays are allocated on arena, so we don't have to free them */
```

We don't have to free them for userland anonymous classes, as mentioned in the comment, I'd assume